sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -113,7 +130,24 @@ public virtual Response PublishEvents(IEnumerable<EventGridEvent> events, Cancel
 
             try
             {
-                return _serviceRestClient.PublishEvents(_hostName, events, cancellationToken);
+                List<EventGridEvent> eventsWithSerializedPayloads = new List<EventGridEvent>();
+                foreach (EventGridEvent egEvent in events)
+                {
+                    EventGridEvent newEGEvent = new EventGridEvent(
+                            egEvent.Id,
+                            egEvent.Subject,
+                            new EventGridSerializer(
+                                egEvent.Data,
+                                _serializer,
+                                cancellationToken),
+                            egEvent.EventType,
+                            egEvent.EventTime,
+                            egEvent.DataVersion);
+                    newEGEvent.Topic = egEvent.Topic;
+
+                    eventsWithSerializedPayloads.Add(newEGEvent);
```

Comment:
nit: factor this out into a `PublishEventsInternal` method taking an `async` parameter.